recipes/Dockerfile

```diff
@@ -0,0 +1,63 @@
+# Run this from the root of the OpenTurbine repository e.g.
+# docker build -f recipes/Dockerfile -t openturbine:latest .
+
+FROM ubuntu:latest
+
+# Use bash as the default shell
+SHELL ["/bin/bash", "-c"]
+
+# Step 1: Install required packages and tools
+RUN apt-get update -qq && \
+    apt-get install -y \
+        build-essential \
+        clang-tidy \
+        cmake \
+        curl \
+        g++ \
+        gfortran \
+        git \
+        libcurl4-openssl-dev \
+        make \
+        python3 \
+        python3-pip \
+        unzip \
+        wget && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Step 2: Get spack and install necessary libraries using it
+RUN git clone https://github.com/spack/spack.git && \
+    source spack/share/spack/setup-env.sh && \
+    spack compiler find && \
+    spack external find && \
+    spack mirror add spack-public https://mirror.spack.io && \
+    spack install googletest && \
+    spack --insecure install cppcheck && \
+    spack install yaml-cpp && \
+    spack --insecure install trilinos@16.0.0~mpi~epetra && \
+    spack --insecure install vtk~mpi~opengl2
+
+# Step 3: Set up the working directory and copy the OpenTurbine repository
+WORKDIR /projects/openturbine
+COPY . .
+
+# Step 4: Load necessary libraries and build the project
+ARG CMAKE_OPTIONS="-DOpenTurbine_ENABLE_SANITIZER_ADDRESS=ON \
```

Comment:
I think the cmake options should mirror what I proposed below for the MacOS script - something production ready for a non-power user to get a build that is ready to deploy.